docs/recipes/build/add-build-script.md

```diff
@@ -1,88 +1,256 @@
+Create a new console app called 'Build' at the root of your solution
 
----
-#### 1. The Build Dependencies
-Paste the following at the end of your `paket.dependencies` file, which is at the root of the solution.
-```
-group Build
-    source https://api.nuget.org/v3/index.json
-    framework: netstandard2.0
-    storage: none
-
-    nuget FSharp.Core
-    nuget Fake.Core.Target
-    nuget Fake.DotNet.Cli
-    nuget Fake.IO.FileSystem
+```fsharp
+dotnet new console -lang f# -n Build
 ```
-Then, execute `paket install` to install these dependencies.
 
-#### 2. Fake CLI Tool
-Run the following commands at the root folder of your solution to install FAKE as a local tool:
-```bash
-dotnet tool install fake-cli
-dotnet tool restore
+#### 2. Create a build script
+
+Open the generated folder.
+
+Open the project in your IDE and rename the file `Program.fs` to `Build.fs`.
+
+> If you just rename the file directly rather than in your IDE, then the Build project won't be able to find it unless you edit the Build.fsproj file as well
+
+Open it and paste in the following code.
+
+```fsharp
+open Fake.Core
+open Fake.IO
+open Farmer
+open Farmer.Builders
+
+open Helpers
+
+initializeContext()
+
+let sharedPath = Path.getFullName "src/Shared"
+let serverPath = Path.getFullName "src/Server"
+let clientPath = Path.getFullName "src/Client"
+let deployPath = Path.getFullName "deploy"
+
+Target.create "Clean" (fun _ -> Shell.cleanDir deployPath)
+
+Target.create "InstallClient" (fun _ -> run npm "install" ".")
+
+Target.create "Bundle" (fun _ ->
+    [ "server", dotnet $"publish -c Release -o \"{deployPath}\"" serverPath
+      "client", dotnet "fable --run webpack -p" clientPath ]
+    |> runParallel
+)
+
+Target.create "Azure" (fun _ ->
+    let web = webApp {
+        name "full"
+        zip_deploy "deploy"
+    }
+    let deployment = arm {
+        location Location.WestEurope
+        add_resource web
+    }
+
+    deployment
+    |> Deploy.execute "full" Deploy.NoParameters
+    |> ignore
+)
+
+Target.create "Run" (fun _ ->
+    run dotnet "build" sharedPath
+    [ "server", dotnet "watch run" serverPath
+      "client", dotnet "fable watch --run webpack-dev-server" clientPath ]
+    |> runParallel
+)
+
+open Fake.Core.TargetOperators
+
+let dependencies = [
+    "Clean"
+        ==> "InstallClient"
+        ==> "Bundle"
+        ==> "Azure"
+
+    "Clean"
+        ==> "InstallClient"
+        ==> "Run"
+]
+
+[<EntryPoint>]
+let main args = runOrDefault args
 ```
 
-#### 3. The Build Script
-Add an F# script file to the root folder of your solution named `build.fsx` and paste the following code into it.
-The script builds and publishes the client and server applications in their production / release modes, and copies the outputs of both into a `deploy` folder at the root of the repository.
+#### 3. Helpers
+
+Add another file to the `Build.fsproj` project called `Helpers.fs` and paste in the following code:
```

Comment:
If possible, can we avoid needing this extra Helpers file?

@theimowski I'm thinking if this could, in the future (please not for v3!), be a separate package or source file somewhere that people could start to pull down.